src/mjolnir/ferry_connections.cc

```diff
@@ -225,13 +225,27 @@ bool ShortFerry(const uint32_t node_index,
   uint64_t wayid = 0;
   bool short_edge = false;
   for (const auto& edge : bundle.node_edges) {
-    // Check ferry edge. If the end node has a non-ferry edge check
-    // the length of the edge
+    // Check ferry edge.
     if (edge.first.attributes.driveable_ferry) {
       uint32_t endnode =
           (edge.first.sourcenode_ == node_index) ? edge.first.targetnode_ : edge.first.sourcenode_;
       auto end_node_itr = nodes[endnode];
       auto bundle2 = collect_node_edges(end_node_itr, nodes, edges);
+
+      // If we notice that either the end node or source node is a connection to another ferry edge,
+      // be cautious and assume this isn't a short edge.
+      for (const auto& edge2 : bundle.node_edges) {
+        if (edge2.first.llindex_ != edge.first.llindex_ && edge2.first.attributes.driveable_ferry) {
+          return false;
+        }
+      }
+      for (const auto& edge2 : bundle2.node_edges) {
+        if (edge2.first.llindex_ != edge.first.llindex_ && edge2.first.attributes.driveable_ferry) {
+          return false;
+        }
+      }
```

Comment:
The "just one edge" check. Looping over all other connected edges at both source and destination nodes. Sees if any of them were also ferry edges, with a check on the latlong index to make sure we're not looking at the original ferry edge. Return `false` immediately if we find one, since we know that means this ferry consists of more than one edge.